### ISSUE TITLE
Aggregated functional abundance tables might contain non-integer values, and this is fine

### ISSUE THREAD

**Issue by fpusan** [jtamames/SqueezeMeta#115]
(Got one email this morning in which a user was worried about the functional abundance tables generated by `sqm2tables.py` and used in `SQMtools` containing non-integer values. I'm copying my answer here in case it helps other users).

In the KEGG and COG databases, some of the references are annotated with more than one KEGG/COG.

So when one of the ORF in the metagenome matches with those references, there are actually two or three KEGG and COG we could use to annotate it. Note that this is not the same as one ORF matching with more than one reference in the database (in that case, if the references had different annotations, we would report the ORF as unclassified due to lack of consensus).

This poses no problem when annotating the ORFs themselves (we just put all the different functions as inherited from the database reference) but it does when aggregating counts to generate functional abundance tables. As a compromise, when we find an ORF annotated to multiple features we split its counts between the different features. This still preserves the sample-wise sum of the matrix (though it introduces non-integer values which I agree are scary when unexpected).

Long story short, it IS an abundance matrix and you can treat it as such. When we truly need an integer count matrix (e.g. for calculating differential abundances with DESeq2) we just round it. Total counts are so high that the rounding error is (in our humble opinion) negligible.
